cpp/client/client.cc

```diff
@@ -26,24 +27,42 @@ Client::~Client() {
 bool Client::Connect() {
   CHECK(!Connected());
 
-  fd_ = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
-  PCHECK(fd_ >= 0) << "Socket creation failed";
-
-  static struct sockaddr_in server_socket;
-  memset(&server_socket, 0, sizeof server_socket);
-  server_socket.sin_family = AF_INET;
-  server_socket.sin_port = htons(port_);
-  CHECK_EQ(1, inet_aton(server_.c_str(), &server_socket.sin_addr))
-      << "Can't parse server address: " << server_;
+  static const addrinfo server_addr_hints = {
+    AI_ADDRCONFIG, /* ai_flags */
+    AF_UNSPEC,     /* ai_family */
+    SOCK_STREAM,   /* ai_socktype */
+    IPPROTO_TCP    /* ai_protocol */
+  };
+
+  struct addrinfo* server_addr;
+
+  int ret = getaddrinfo(server_.c_str(), port_.c_str(), &server_addr_hints,
+                        &server_addr);
```

Comment:
As far as I can see, this class is only ever used by SSLClient, which in turn is only used by cpp/client/ct.cc - a command-line tool. I think allowing that to block for DNS lookups is a reasonable trade-off in exchange for allowing hostnames.